api/api_test.go

```diff
@@ -303,6 +304,57 @@ func TestAPIResolve(t *testing.T) {
 	}
 }
 
+// TestRNSResolve tests resolving content from RNS addresses
+func TestRNSResolve(t *testing.T) {
+	rnsAddr := "marcelosdomain.rsk"
+	resolvedContent := "88ced8ba8e9396672840b47e332b33d6679d9962d80cf340d3cf615db23d4e07"
+
+	type test struct {
+		desc        string
+		ctx         context.Context
+		addr        string
+		content     string
+		expectedErr error
+	}
+
+	tests := []*test{
+		{
+			desc:        "valid RSK domain",
+			addr:        rnsAddr,
+			content:     resolvedContent,
+			expectedErr: nil,
+		},
+		{
+			desc:        "invalid RSK domain",
+			addr:        ".rsk",
+			content:     resolvedContent,
+			expectedErr: rns.ErrNoContent,
+		},
+	}
+
+	for _, x := range tests {
+		t.Run(x.desc, func(t *testing.T) {
+			api := NewAPI(nil, nil, rns.ResolveDomainContent, nil, nil, nil)
```

Comment:
Will rns.ResolveDomainContent actually make a resolve request on the external service? If that is the case, the provided RNS resolver should be mocked to test only the functionality of the api Resolve function.

Reply:
I agree, we changed the test with a mock of rns, in a similar manner to the ens mock.